### PR TITLE
Also check for lowercase `location` header on redirect

### PR DESCRIPTION
**Pull request by denniss17**
Fixes https://github.com/kittinunf/Fuel/issues/279

When there is a redirect status, fuel will first detect the default `Location` header. If it is not present, it will try to use the `location` header.

I looked into creating a test for this, but I did not see a simple way to make https://httpbin.org/ return a redirect with a custom header.

### CODE REVIEW

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ fun redirectResponseInterceptor(manager: FuelManager) =
                         response.statusCode == HttpsURLConnection.HTTP_MOVED_TEMP ||
                         response.statusCode == 307   // 307 TEMPORARY REDIRECT - https://httpstatuses.com/307
                         ) {
-                    val redirectedUrl = response.headers["Location"]
+                    val redirectedUrl? = response.headers.containsKey("Location") ? response.headers["Location"] : response.headers["location"]
```
Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt]: Do you think if we could use `inline fun String.toUpperCase(): String` or `lowercase` counterpart to simplify our logic a bit?

Comment (kindraywind) [reply fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt]: We're almost there! @denniss17 
Let's change a minor thing 👍

Comment (denniss17) [reply fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt]: No, except when we lowercase all keys in the headers map

**Comment by kittinunf** [comment]
I think this is reasonable. Merging this in.

**Pull request merged**